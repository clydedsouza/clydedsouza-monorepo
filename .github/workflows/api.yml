name: Build and deploy API

on:
  push:
    branches:
      - main
    paths:
      - "api/**"
  pull_request:
    branches:
      - main
    paths:
      - "api/**"
  workflow_dispatch:

jobs:
  build_api:
    name: Build API
    runs-on: ubuntu-latest
    environment:
      name: api
    defaults:
      run:
        shell: bash
        working-directory: api

    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Install Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18.x

      - name: Install NPM packages
        run: npm ci

      - name: Generate API files
        run: npm run generate:api

      - name: Upload API files
        uses: actions/upload-artifact@v4
        with:
          path: ./api/public
          include-hidden-files: true

  deploy_api:
    name: Deploy API
    runs-on: ubuntu-latest
    environment:
      name: api
    needs: build_api
    if: |
      success() &&
      github.ref == 'refs/heads/main'
    defaults:
      run:
        shell: bash

    steps:
      - name: Download artifact
        uses: actions/download-artifact@v4
        with:
          name: artifact
          path: artifact

      - name: Deploy artifacts to the API repo
        uses: tagus/git-deploy@v0.4.1
        with:
          changes: artifact
          repository: git@github.com:clydedsouza/clydedsouza-web-api.git
          ssh_key: ${{ secrets.API_REPO_DEPLOY_KEY }}
          name: clydedsouza
          email: 16275549+clydedsouza@users.noreply.github.com
          clean_repo: true
          branch: main
          message: "Commit from the monorepo API deploy job"
