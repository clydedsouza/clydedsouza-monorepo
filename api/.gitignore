# Dependency directories
node_modules/

# Optional npm cache directory
.npm

# Exclude temp 'working' directory
public/*.json