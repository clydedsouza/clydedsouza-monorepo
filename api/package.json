{
  "name": "clydedsouza-projects",
  "version": "0.0.1",
  "author": "Clyde D'Souza",
  "scripts": {
    "generate:portfolio": "markdown-json -c ./settings/portfolio.json",
    "generate:projects": "markdown-json -c ./settings/projects.json",
    "generate:books": "markdown-json -c ./settings/books.json",    
    "generate:speaking": "markdown-json -c ./settings/speaking.json",
    "generate:teaching": "markdown-json -c ./settings/teaching.json",
    "generate:highlights": "markdown-json -c ./settings/highlights.json",
    "generate:pinned": "markdown-json -c ./settings/pinned.json",
    "generate:api": "npm run generate:projects & npm run generate:teaching & npm run generate:books & npm run generate:speaking & npm run generate:highlights & npm run generate:pinned & npm run generate:portfolio"
  },
  "dependencies": {
    "markdown-json": "^1.6.1"
  }
}
