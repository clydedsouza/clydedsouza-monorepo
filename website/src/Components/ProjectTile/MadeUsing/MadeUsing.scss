.made-using {
  span {
    font-size: small;
    background-color: #ffffff30;
    color: white;
    padding: 1px 9px;
    border-radius: 10px;
    display: inline-block;
    margin: 9px 3px 0 0;
    font-weight: 100;
  }
}
