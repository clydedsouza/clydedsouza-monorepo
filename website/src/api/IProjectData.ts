export interface IProjectData {
  data: IProject[]
}

export interface IProject {
  id: string
  title: string
  date: string
  description: string
  website: string
  github: string
  madeUsing: string[]
  category: string
  image: string
  imageDescription: string
}
