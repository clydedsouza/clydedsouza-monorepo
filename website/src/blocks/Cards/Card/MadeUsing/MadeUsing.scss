.made-using {
  margin: 12px 0;

  span {
    font-size: small;
    background-color: #ffffff30;
    color: white;
    padding: 1px 9px;
    border-radius: 10px;
    display: inline-block;
    margin: 3px;
    font-weight: 100;
  }
}
