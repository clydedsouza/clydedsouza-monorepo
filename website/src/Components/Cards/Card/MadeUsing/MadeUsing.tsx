import { IProject } from '../../../../Api/IProjectData'
import './MadeUsing.scss'

function MadeUsing(props: Partial<IProject>) {
  if (!props.madeUsing || props.madeUsing.length < 1) return <></>

  return (
    <div className="made-using">
      {props.madeUsing &&
        props.madeUsing.map((item) => <span key={item}>{item}</span>)}
    </div>
  )
}

export default MadeUsing
