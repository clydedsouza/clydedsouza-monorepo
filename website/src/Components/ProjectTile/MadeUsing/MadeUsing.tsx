import { IProject } from '../../../Types/ProjectData'
import './MadeUsing.scss'

function MadeUsing(props: IProject) {
  return (
    <div className="made-using">
      {props.madeUsing &&
        props.madeUsing.map((item) => <span key={item}>{item}</span>)}
    </div>
  )
}

export default MadeUsing
