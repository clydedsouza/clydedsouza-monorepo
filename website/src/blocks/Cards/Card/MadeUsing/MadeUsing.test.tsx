import { IProject } from "@/api/IProjectData";
import { render } from "@testing-library/react";
import MadeUsing from "./MadeUsing";

describe("Made using", () => {
  it.each([
    {
      madeUsing: [],
    },
    {
      madeUsing: ["html"],
    },
    {
      madeUsing: ["html", "javascript"],
    },
  ])(
    "should render made using component when $madeUsing is supplied",
    ({ madeUsing }) => {
      const project: Partial<IProject> = {
        madeUsing,
      };
      const { container } = render(<MadeUsing {...project} />);
      expect(container).toMatchSnapshot();
    }
  );
});
