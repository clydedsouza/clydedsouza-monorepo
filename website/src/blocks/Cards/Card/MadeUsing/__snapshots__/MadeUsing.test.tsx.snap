// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Made using should render made using component when ["html", "javascript"] is supplied 1`] = `
<div>
  <div
    class="made-using"
  >
    <span>
      html
    </span>
    <span>
      javascript
    </span>
  </div>
</div>
`;

exports[`Made using should render made using component when ["html"] is supplied 1`] = `
<div>
  <div
    class="made-using"
  >
    <span>
      html
    </span>
  </div>
</div>
`;

exports[`Made using should render made using component when [] is supplied 1`] = `<div />`;
